docs/developer/api_guidelines.md

```diff
@@ -0,0 +1,438 @@
+* **should** return the resource id in the *Location* header as a relative URL if it was used for creating of a new resource
+* **should** return a response body that contains the resource if it was used for creating of a new resource
+* **must** return an empty response body a HTTP 204 -No content if it was used for updating an existing resource
+
+## partial update
+Used do a partial update on a resource
+* **must** map to HTTP PATCH method
+* **must** contain updated fields in JSON Merge Patch format (https://datatracker.ietf.org/doc/html/rfc7396) in the request body
+* **must** not be used to create a new resource
+* **must** return HTTP 204 -No Content on success
+* **must** return an empty response body
+
+## delete
+Used to remove a resource
+* **must** map to HTTP DELETE method
+* **may** be applicable to a single resource or a collection of resources. A filter criteria **must** be provided as query parameters to delete a collection of resources.
+* **must** not use a request body
+* **must** not use a response body
+* **must** return HTTP 204 - No Content on success
+* **must** return HTTP 202 - Accepted on async. deletion request
+
+---
+# Time Values Conventions
+
+## Time Formats
+
+Time values **should** be represented as UTC times and **must** be encoded according to the [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339) standard.
+This standard accepts multiple variations of time formats, however to avoid confustion all time values in our APIs **should** be stored and represented using the following format:
+
+- `2022-06-29T08:56:38.547Z`
+
+When ingesting time values, i.e. by accepting them as request parameters, APIs **should** be able to handle multiple variations of the RFC3339 standard. Examples:
+
+- `1985-04-12T23:20:50.52Z`
+- `1996-12-19T16:39:57-08:00`
+- `1990-12-31T23:59:60Z`
+- `1990-12-31T15:59:60-08:00`
+- `1937-01-01T12:00:27.87+00:20`
+- `2022-06-28T13:43:14.71657055Z`
+
+## Time Value Property Naming
+
+All properties referring to a time value **must** end with the suffix `Time`, e.g. `modifyTime`, `triggerTime` etc. This suffix shall be used consistently across all APIs, meaning that
+we **should** not have properties like `creationTimestamp`, `modifiedAt` etc.
+For time frames, the property names `startTime` and `endTime` **should** be consistently used - other variations like `fromTime` or `toTime` **should** therefore NOT be used. 
+
+---
+# Versioning
+
+Each service has its own individual API version, i.e., there is no shared version for all services within Keptn. The versioning scheme **must** follow
+the semantic versioning conventions by maintaining a `major`, `minor` and `patch` version (`<major>.<minor>.<patch>`).
+
+These versions **must** be incremented according to the following rules:
+
+## Major Version Updates
+The major version is increased whenever there is a breaking change, such as a modified structure of a request model, or changed HTTP status codes.
```

Comment:
How to handle the following case? 
I perform a `GET /project/nonexisting-project` and I get a 500. Since the project does not exist, the expected HTTP return code should be 404. Proposing the bugfix should we bump the version or not? In case, which part of the version should be bumped?